src/app/auth/auth.component.ts

```diff
@@ -170,11 +170,8 @@ export class AuthComponent implements OnInit, OnDestroy {
     this.githubService.storeOrganizationDetails(org, dataRepo);
 
     this.phaseService.storeSessionData().pipe(
-      map((isValidSession: boolean) => {
-        if (!isValidSession) {
-          throw new Error('Invalid Session');
-        }
-      })
+      filter(isValidSession => isValidSession),
+      throwIfEmpty(() => new Error('Invalid Session'))
```

Comment:
This might be OK for now but in the future if there are multiple conditions which could lead to failure, would this pattern still work?

Reply:
To handle multiple conditions, we could generalise this pattern. We can create our own operator `throwIfFalse`, that could be used instead of this `filter` + `throwIfEmpty` combination?

Reply:
Okay sure, thanks!

Reply:
I've created a new operator that tests a number of conditions on the SessionData object.
I think this makes the session validation code more readable, what do you think?